fuel-vm/src/tests/blockchain.rs

```diff
@@ -2261,3 +2261,444 @@ fn various_ldc_issues_poc() {
         }
     });
 }
+
+// === Begin temporary tests for ldcv2 ===
+
+#[test]
+fn ldcv2__offset_affects_read_code() {
```

Comment:
How does it affect it? Based on the test, it looks like as offset increases, the gas used decreases. G/W/T would help clarify the name and vice versa.

It looks like a lot of the tests in this suite are missing G/W/T and names that capture the specific intended behavior.

For example, `ldcv2_ssp_not_sp` doesn't say what should be `ssp`, or in what scenario, just that it's not `sp`. Shouldn't it be something like `ldcv2__when_increase_sp_...something`

Reply:
Really I just copied the tests for original LDC and adapted them when necessary. The original cases for LDC should be fixed as well.

The naming is indeed unclear, I'll try to fix those. I consider G/W/T unhelpful, but I can add format the tests that way if you want to.

Reply:
> The naming is indeed unclear, I'll try to fix those. 

Cool. 

> I consider G/W/T unhelpful, but I can add format the tests that way if you want to.

From my perspective, each test should be only be testing one thing. I find G/W/T to be helpful to quickly parsing the tests to see what's being tested in that specific test. As I scroll down a test suite, I expect there to be many tests that do similar but subtly different things. 

`#[test_case]` also allows that, but I think a lot of engineers try to make it do too much in the name of DRY. So I prefer G/W/T clarity over `test_case` brevity.

Reply:
I added G/W/T where applicable. I'm not happy with that, especailly since the distinction between Given and When is quite tricky when data is code.